new Leetcode/36. Valid Sudoku/36. Valid Sudoku.js

```diff
@@ -0,0 +1,59 @@
+/**
+ * @param {character[][]} board
+ * @return {boolean}
+ */
+var isValidSudoku = function (board) {
+    if (!board || board.length < 9 || board[0].length < 9) {
+
+        return false
+    }
+    var mapping = {};
+    for (var i = 0; i < 9; i++) {
+        for (var j = 0; j < 9; j++) {
+            if (board[i][j] !== '.') {
+                if (mapping[board[i][j]]) {
+                    return false;
+                } else {
+                    mapping[board[i][j]] = true;
+                }
+            }
+        }
+    }
+    mapping = {};
+    for (i = 0; i < 9; i++) {
+        for (j = 0; j < 9; j++) {
+            if (board[j][i] !== '.') {
+                if (mapping[board[j][i]]) {
+                    return false;
+                } else {
+                    mapping[board[j][i]] = true;
+                }
+            }
+        }
+    }
+
+    for (var m = 0; m < 9; m++) {
```

Comment:
another idea is to write 4 for loops. `for (m = 0; m < 3; m++) { for (n = 0; n < 3; n++) {  blah blah } }`